pdns/iputils.hh

```diff
@@ -884,17 +884,20 @@ private:
     }
   }
 
-  void copyTree(const NetmaskTree& rhs)
-  {
-    TreeNode *node;
-
-    node = rhs.d_root.get();
-    if (node != nullptr)
-      node = node->traverse_l();
-    while (node != nullptr) {
-      if (node->assigned)
-        insert(node->node.first).second = node->node.second;
-      node = node->traverse_lnr();
+  void copyTree(const NetmaskTree& rhs) noexcept
+  {
+    try {
+      TreeNode *node = rhs.d_root.get();
+      if (node != nullptr)
+        node = node->traverse_l();
+      while (node != nullptr) {
+        if (node->assigned)
+          insert(node->node.first).second = node->node.second;
+        node = node->traverse_lnr();
+      }
+    }
+    catch (...) {
+      abort();
```

Comment:
I agree that copying an existing tree is probably safe in most cases, but calling `abort()` if we run out of memory while trying to copy a large NetmaskTree seems wrong?

Reply:
I  am not very happy with that either, reading up on the exact semantics of `noexcept' to see if we can only catch `NetmaskException` here.

Reply:
If I understand correctly, `noexcept` fucntions that actually throw cause a runtime kill.
So `noexcept` cannot be used if we cacth `NetmaskException` only, as allocation failures might still throw. Calling code would have a chance to react on OOM situations. Let's see if Coverity likes that enough.